File: Lab4/C#/Vidrizok.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab4
{
    public class Vidrizok
    {
        Tochka start;
        Tochka end;
        public double Dlina { get; private set; }

        public Vidrizok() : this(new Tochka(), new Tochka()) { }
        public Vidrizok(Tochka s, Tochka e)
        {
            start = s;
            end = e;
            Getdlina();
        }
        public Vidrizok(double x1, double y1, double x2, double y2)
        {

            start = new Tochka(x1, y1);
            end = new Tochka(x2, y2);
            Getdlina();
        }
        public Vidrizok(Vidrizok obj) : this(obj.start, obj.end) { }
        public Vidrizok(Tochka s, double x2, double y2)
        {
            start = s;
            end = new Tochka(x2, y2);
            Getdlina();
        }
        public Vidrizok(double x1, double y1, Tochka e)
        {
            start = new Tochka(x1, y1);
            end = e;
            Getdlina();
        }
        void Getdlina()
        {
            Dlina = Math.Sqrt(Math.Pow(start.x - end.x, 2) + Math.Pow(start.y - end.y, 2));
        }
        public static Vidrizok operator +(Vidrizok obj1, Vidrizok obj2)
        {
            Vidrizok result = new Vidrizok();
            result.start = obj1.start + obj2.start;
            result.end = obj1.end + obj2.end;
            result.Getdlina();
            return result;
        }
        public static Vidrizok operator -(Vidrizok obj1, Vidrizok obj2)
        {
            Vidrizok result = new Vidrizok();
            result.start = obj1.start - obj2.start;
            result.end = obj1.end - obj2.end;
            result.Getdlina();
            return result;
        }
        public void Sdvig(double x, double y)
        {
            this.start = this.start + new Tochka(x, y);
            this.end = this.end + new Tochka(x, y);
            this.Getdlina();
        }
    }
}

File: Lab4/C++/Vidrizok.cpp
#pragma once
#include "Tochka.cpp"
#include <string>

using namespace std;

class Vidrizok
{
		Tochka start;
		Tochka end;
		public: 
			double Dlina;

			Vidrizok() { start = Tochka(); end = Tochka(); }
		Vidrizok(Tochka s, Tochka e)
		{
			start = s;
			end = e;
			Getdlina();
		}
		Vidrizok(double x1, double y1, double x2, double y2)
		{

			start = Tochka(x1, y1);
			end = Tochka(x2, y2);
			Getdlina();
		}
		Vidrizok(Vidrizok &obj) 
		{ 
			start = obj.start; 
			end = obj.end; 
		}
		Vidrizok(Tochka s, double x2, double y2)
		{
			start = s;
			end = Tochka(x2, y2);
			Getdlina();
		}
		Vidrizok(double x1, double y1, Tochka e)
		{
			start = Tochka(x1, y1);
			end = e;
			Getdlina();
		}
		Vidrizok operator +(Vidrizok obj2)
		{
			Vidrizok result;
			result.start = start + obj2.start;
			result.end = end + obj2.end;
			result.Getdlina();
			return result;
		}
		Vidrizok operator -(Vidrizok obj2)
		{
			Vidrizok result;
			result.start = start - obj2.start;
			result.end = end - obj2.end;
			result.Getdlina();
			return result;
		}
		void Sdvig(Vidrizok &obj, double x, double y)
		{
			Tochka temp (x, y);
			obj.start = obj.start + temp;
			obj.end = obj.end + temp;
			obj.Getdlina();
		}

		private: void Getdlina()
		{
			Dlina = sqrt(pow(start.x - end.x, 2) + pow(start.y - end.y, 2));
		}
		
};

